.github/workflows/ci.yml

```diff
@@ -0,0 +1,48 @@
+name: CI Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  GOLANGCI_LINT_VERSION: v1.61.0
+
+jobs:
+  build-lint-test:
+    name: CI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Setup Mage
+        uses: magefile/mage-action@v3
+        with:
+          install-only: true
+
+      - name: Install test dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y gcc
+
+      - name: Setup CI Linter
+        run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin ${{env.GOLANGCI_LINT_VERSION}}
+
+      - name: Build
+        run: |
+          mage -v build
+
+      - name: Lint
+        run: |
+          mage -v lint
```

Comment:
Do we want to use the available action perhaps like in Grafana, https://github.com/grafana/grafana/blob/47115c714adfcf07c1898021581d5f4a817f9c45/.github/workflows/go_lint.yml#L26? No strong opinions, but feels perhaps easier to update it whenever needed. I guess we don't have a good story currently enforcing the same version locally and in CI, but something we can improve eventually perhaps. @xnyo any opinion?

Reply:
Yup, we can definitely use the https://github.com/golangci/golangci-lint-action. Just needs to be in a separate job/action as recommended in the docs.

Reply:
Tried with golangci-lint actions but weirdly keeps failing with
```
  Running [/home/runner/golangci-lint-1.61.0-linux-amd64/golangci-lint run  ./...] in [/home/runner/work/grafana-plugin-sdk-go/grafana-plugin-sdk-go] ...
  level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
  
  Error: golangci-lint exit with code 4
  Ran golangci-lint in 63034ms
```

Reply:
takes around 7 mins to finish. Ready for review.